pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -197,6 +205,16 @@ public String segmentCommitEnd(@QueryParam(SegmentCompletionProtocol.PARAM_INSTA
       return SegmentCompletionProtocol.RESP_FAILED.toJsonString();
     }
 
+    SegmentMetadataImpl segmentMetadata;
+    try {
+      segmentMetadata = extractMetadataFromSegmentFile(segmentName, new java.net.URI(segmentLocation));
```

Comment:
Can you please update the rest of the file to use java.net.URI as well? (and therefore import the right class and not qualify it everywhere)? It should be ok since the server only treats this as a string, so no compat issues.

Reply:
Use java.net.URI uniformly now. In fact, my test shows that the original apache uri usage may not work for URI with % char in it. Please see the two added tests in LLCRealtimeClusterSplitCommitIntegrationTest() for details. They can be deleted when this PR is submitted.